### PR TITLE
Oauth bearer header

### LINKED ISSUE #93: Add missing headers to OAuth JWT bearer token
The JWT MUST have:

- typ
- alg
- x5c

headers

### PR DESCRIPTION
**Pull request by woutslakhorst**
Closes #93

### CODE REVIEW

File: pkg/jwx.go [hunk 1]
```diff
@@ -63,7 +63,30 @@ func (client *Crypto) SignJWT(claims map[string]interface{}, key types.KeyIdenti
 		return "", err
 	}
 
-	token, err = SignJWT(rsaPrivateKey, claims)
+	token, err = SignJWT(rsaPrivateKey, claims, nil)
+	return
+}
+
+// SignJWTRFC003 sign a JWT according to Nuts RFC003. This func is only for signing the bearer token of the oauth flow.
```
Comment (reinkrul) [inline pkg/jwx.go]: sign -> signs

File: pkg/jwx.go [hunk 2]
```diff
@@ -63,7 +63,30 @@ func (client *Crypto) SignJWT(claims map[string]interface{}, key types.KeyIdenti
...
+func (client *Crypto) SignJWTRFC003(claims map[string]interface{}, key types.KeyIdentifier) (token string, err error) {
```
Comment (reinkrul) [inline pkg/jwx.go]: we're providing a key parameter but not using it? bit confusing. If you really don't want to use it you can call it udnerscore (`_`)

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: pkg/jwx.go (hunk 2 above)
Comment (woutslakhorst) [reply pkg/jwx.go]: oversight

File: pkg/jwx.go [hunk 3]
```diff
@@ -63,7 +63,30 @@ func (client *Crypto) SignJWT(claims map[string]interface{}, key types.KeyIdenti
...
+	var (
+		certificate *x509.Certificate
+		privateKey  crypto.Signer
+	)
+	if certificate, privateKey, err = client.generateVendorEphemeralSigningCertificate(); err != nil {
+		return
+	}
+
+	if err != nil {
+		return "", err
```
Comment (reinkrul) [inline pkg/jwx.go]: unreachable code (and "" is the default value for a string since it can't be nil)

**Review by reinkrul — CHANGES_REQUESTED** [review]
Sorry just one last nag :-p

**Comment by woutslakhorst** [comment]
@reinkrul I also changed the JWT signing from RS256 to PS256. This was causing problems downstream.

**Review by reinkrul — APPROVED** [review]

**Pull request merged**